diff --git a/xapian-core/Makefile.am b/xapian-core/Makefile.am
index f513c5fd0..57c5aaed1 100644
--- a/xapian-core/Makefile.am
+++ b/xapian-core/Makefile.am
@@ -102,11 +102,11 @@ dist_m4data_DATA = m4-macros/xapian-1.5.m4
 
 # Install the pkg-config file containing the correct compiler options.
 pkgconfigdir = $(libdir)/pkgconfig
-nodist_pkgconfig_DATA = pkgconfig/xapian-core-1.5.pc
-DISTCLEANFILES += pkgconfig/xapian-core-1.5.pc
+nodist_pkgconfig_DATA = pkgconfig/xapian-core.pc
+DISTCLEANFILES += pkgconfig/xapian-core.pc
 
-libxapian_la = libxapian-1.5.la
-lib_LTLIBRARIES = libxapian-1.5.la
+libxapian_la = libxapian.la
+lib_LTLIBRARIES = libxapian.la
 libxapian_1_5_la_SOURCES = $(lib_src)
 libxapian_1_5_la_LIBADD = $(XAPIAN_LIBS) $(ICU_LIBS)
 libxapian_1_5_la_LDFLAGS = \
diff --git a/xapian-core/configure.ac b/xapian-core/configure.ac
index e0fe1b432..1cbc2dcc1 100644
--- a/xapian-core/configure.ac
+++ b/xapian-core/configure.ac
@@ -36,20 +36,6 @@ AC_SUBST([LIBRARY_VERSION_INFO])
 LIBRARY_VERSION_SUFFIX=
 dnl Where xapian.h, etc go.
 incdir=$includedir
-case $PACKAGE_VERSION in
-  [*.*[13579].*])
-    dnl Development release series:
-    [dev_series=`echo "$PACKAGE_VERSION"|sed 's/\.[^.]*$//'`]
-
-    dnl Append version installed programs by default.  To specify no suffix,
-    dnl configure with: --program-suffix=
-    test x"$program_suffix" != xNONE || program_suffix=-$dev_series
-
-    incdir=$incdir/xapian-$dev_series
-
-    LIBRARY_VERSION_SUFFIX=-$dev_series
-    ;;
-esac
 
 AC_SUBST([LIBRARY_VERSION_SUFFIX])
 AC_SUBST([incdir])
