commit fa097f60e0e22d079b2784c2a5f165d85d10e221
Author: ttyS3 <ttys3.rust@gmail.com>
Date:   2024-09-25 02:06:36 +0000

    EnlargeObjectSize.patch

diff --git a/Source/bmalloc/libpas/src/chaos/Chaos.cpp b/Source/bmalloc/libpas/src/chaos/Chaos.cpp
index 03721979..00c957e3 100644
--- a/Source/bmalloc/libpas/src/chaos/Chaos.cpp
+++ b/Source/bmalloc/libpas/src/chaos/Chaos.cpp
@@ -51,7 +51,7 @@ namespace {
     macro(uintptr_t,     seed,                           666,             "%lu") \
     macro(uintptr_t,     seedAdvance,                    1,               "%lu") \
     macro(uintptr_t,     maxSmallObjectSize,             1000,            "%lu") \
-    macro(uintptr_t,     maxObjectSize,                  65536,           "%lu") \
+    macro(uintptr_t,     maxObjectSize,                  2097152,         "%lu") \
     macro(double,        smallObjectProbability,         .95,             "%lf") \
     macro(double,        memalignProbability,            .05,             "%lf") \
     macro(uintptr_t,     maxMemalignShift,               14,              "%lu") \
