commit 133d4fd3b8627c934771dc50fd9dcc7e948a7b4f

diff --git a/quickjs.c b/quickjs.c
index 4e58a98..9e46a12 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -55544,3 +55544,14 @@ void JS_AddIntrinsicTypedArrays(JSContext *ctx)
     JS_AddIntrinsicAtomics(ctx);
 #endif
 }
+
+JSClassID JS_GetClassID(JSValueConst v)
+{
+    JSObject *p;
+
+    if (JS_VALUE_GET_TAG(v) != JS_TAG_OBJECT)
+        return 0;
+    p = JS_VALUE_GET_OBJ(v);
+    assert(p != 0);
+    return p->class_id;
+}
diff --git a/quickjs.h b/quickjs.h
index 700ee61..4cc0d2b 100644
--- a/quickjs.h
+++ b/quickjs.h
@@ -500,6 +500,7 @@ typedef struct JSClassDef {
 } JSClassDef;
 
 JSClassID JS_NewClassID(JSClassID *pclass_id);
+JSClassID JS_GetClassID(JSValueConst v);
 int JS_NewClass(JSRuntime *rt, JSClassID class_id, const JSClassDef *class_def);
 int JS_IsRegisteredClass(JSRuntime *rt, JSClassID class_id);
 
