commit a2a90effdc33073e28102c0f90b762565cbf357d

diff --git a/quickjs.c b/quickjs.c
index 9e46a12..5dc3923 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -7898,6 +7898,8 @@ static JSValue JS_GetPropertyValue(JSContext *ctx, JSValueConst this_obj,
         uint32_t idx;
         /* fast path for array access */
         p = JS_VALUE_GET_OBJ(this_obj);
+        if (unlikely(!p->fast_array))
+            goto slow_path;
         idx = JS_VALUE_GET_INT(prop);
         switch(p->class_id) {
         case JS_CLASS_ARRAY:
