--- doxygen-1.9.6.orig/templates/latex/latexmakefile.tpl	2023-03-09 10:29:08.772377871 +0100
+++ doxygen-1.9.6/templates/latex/latexmakefile.tpl	2023-03-09 10:29:23.754555476 +0100
@@ -9,7 +9,7 @@
 {# TODO: generateBib #}
 	pdflatex refman
 	latex_count=8 ; \
-	while egrep -s 'Rerun (LaTeX|to get cross-references right)' refman.log && [ $$latex_count -gt 0 ] ;\
+	while grep -E -s 'Rerun (LaTeX|to get cross-references right)' refman.log && [ $$latex_count -gt 0 ] ;\
 	    do \
 	      echo "Rerunning latex...." ;\
 	      pdflatex refman ;\
@@ -43,7 +43,7 @@
 	echo "Rerunning latex...."
 	{{ config.LATEX_CMD_NAME }} refman.tex
 	latex_count=8 ; \
-	while egrep -s 'Rerun (LaTeX|to get cross-references right)' refman.log && [ $$latex_count -gt 0 ] ;\
+	while grep -E -s 'Rerun (LaTeX|to get cross-references right)' refman.log && [ $$latex_count -gt 0 ] ;\
 	    do \
 	      echo "Rerunning latex...." ;\
 	      {{ config.LATEX_CMD_NAME }} refman.tex ;\
--- doxygen-1.9.6.orig/src/latexgen.cpp	2023-03-09 11:10:52.092688592 +0100
+++ doxygen-1.9.6/src/latexgen.cpp	2023-03-09 11:11:07.666867516 +0100
@@ -346,7 +346,7 @@
     t << "\techo \"Rerunning latex....\"\n"
       << "\t$(LATEX_CMD) $(MANUAL_FILE).tex\n"
       << "\tlatex_count=$(LATEX_COUNT) ; \\\n"
-      << "\twhile egrep -s 'Rerun (LaTeX|to get cross-references right|to get bibliographical references right)' $(MANUAL_FILE).log && [ $$latex_count -gt 0 ] ;\\\n"
+      << "\twhile grep -E -s 'Rerun (LaTeX|to get cross-references right|to get bibliographical references right)' $(MANUAL_FILE).log && [ $$latex_count -gt 0 ] ;\\\n"
       << "\t    do \\\n"
       << "\t      echo \"Rerunning latex....\" ;\\\n"
       << "\t      $(LATEX_CMD) $(MANUAL_FILE).tex ; \\\n"
@@ -374,7 +374,7 @@
     }
     t << "\t$(LATEX_CMD) $(MANUAL_FILE)\n"
       << "\tlatex_count=$(LATEX_COUNT) ; \\\n"
-      << "\twhile egrep -s 'Rerun (LaTeX|to get cross-references right|to get bibliographical references right)' $(MANUAL_FILE).log && [ $$latex_count -gt 0 ] ;\\\n"
+      << "\twhile grep -E -s 'Rerun (LaTeX|to get cross-references right|to get bibliographical references right)' $(MANUAL_FILE).log && [ $$latex_count -gt 0 ] ;\\\n"
       << "\t    do \\\n"
       << "\t      echo \"Rerunning latex....\" ;\\\n"
       << "\t      $(LATEX_CMD) $(MANUAL_FILE) ;\\\n"
