diff --git a/cmake/Findxapian.cmake.old b/cmake/Findxapian.cmake
index 33726cf..79790db 100644
--- a/cmake/Findxapian.cmake.old
+++ b/cmake/Findxapian.cmake
@@ -14,9 +14,9 @@ if(XAPIAN_INCLUDE_DIR AND XAPIAN_LIBRARIES)
     set(xapian_FIND_QUIETLY TRUE)
 endif(XAPIAN_INCLUDE_DIR AND XAPIAN_LIBRARIES)
 
-FIND_PATH(XAPIAN_INCLUDE_DIR xapian/version.h)
+FIND_PATH(XAPIAN_INCLUDE_DIR xapian-1.5/xapian/version.h)
 
-FIND_LIBRARY(XAPIAN_LIBRARIES NAMES xapian)
+FIND_LIBRARY(XAPIAN_LIBRARIES NAMES xapian-1.5)
 
 IF(XAPIAN_INCLUDE_DIR AND XAPIAN_LIBRARIES)
    SET(XAPIAN_FOUND TRUE)
